server/storage.go

```diff
@@ -561,3 +563,105 @@ func saveGDriveToken(path string, token *oauth2.Token, logger *log.Logger) {
 
 	json.NewEncoder(f).Encode(token)
 }
+
+type StorjStorage struct {
+	Storage
+	uplink  *uplink.Uplink
+	project *uplink.Project
+	bucket  *uplink.Bucket
+	logger  *log.Logger
+}
+
+func NewStorjStorage(endpoint, apiKey, bucket, encKey string, logger *log.Logger) (*StorjStorage, error) {
+	var instance StorjStorage
+	var err error
+
+	ctx := context.Background()
+
+	instance.uplink, err = uplink.NewUplink(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new Uplink Instance: %v", err)
+	}
+
+	key, err := uplink.ParseAPIKey(apiKey)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse api key %v", err)
+	}
+
+	instance.project, err = instance.uplink.OpenProject(ctx, endpoint, key)
+	if err != nil {
+		return nil, fmt.Errorf("could not open project: %v", err)
+	}
+
+	access := uplink.NewEncryptionAccessWithDefaultKey(toStorjKey(encKey))
```

Comment:
Is it intentional to use `toStorjKey` instead of `SaltedKeyFromPassphrase`?

Reply:
No, i didnt knew there is a better way, do you have an exmaple?

Reply:
https://github.com/storj/storj/blob/master/lib/uplink/ex_password_test.go

Reply:
Okay fixed.